include/nudb/native_file.hpp

```diff
@@ -8,68 +8,28 @@
 #ifndef NUDB_NATIVE_FILE_HPP
 #define NUDB_NATIVE_FILE_HPP
 
-#include <nudb/error.hpp>
-#include <nudb/file.hpp>
-#include <nudb/posix_file.hpp>
-#include <nudb/win32_file.hpp>
-#include <string>
 
-namespace nudb {
-
-/** A native file handle.
+#if ! GENERATING_DOCS
 
-    This type is set to the appropriate platform-specific
-    implementation to meet the file wrapper requirements.
-*/
-using native_file =
 #ifdef _MSC_VER
-    win32_file;
+#include <nudb/win32_file.hpp>
 #else
-    posix_file;
+#include <nudb/posix_file.hpp>
 #endif
```

Comment:
For me the question is: would someone building on a windows platform every want to use posix_file.hpp ?  Or vice versa? I don't think it would even compile.  To address this, you've include the idea of native_file. I see the merit in this.  But even assuming one could compile both in - which I'm not convinced of - why would want to do this?  I've got confess I've been confused by this from the start and still don't get it.  The documentation reflects - for better or worse - the understanding of an experienced person approaching the library with an open mind.